x-pack/plugin/ml/log-structure-finder/src/main/java/org/elasticsearch/xpack/ml/logstructure/SeparatedValuesLogFileStructureFinder.java

```diff
@@ -0,0 +1,458 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.ml.logstructure;
+
+import org.elasticsearch.common.collect.Tuple;
+import org.elasticsearch.xpack.ml.logstructure.TimestampFormatFinder.TimestampMatch;
+import org.supercsv.exception.SuperCsvException;
+import org.supercsv.io.CsvListReader;
+import org.supercsv.prefs.CsvPreference;
+import org.supercsv.util.Util;
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.DoubleSummaryStatistics;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Random;
+import java.util.SortedMap;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class SeparatedValuesLogFileStructureFinder extends AbstractStructuredLogFileStructureFinder implements LogFileStructureFinder {
+
+    private static final int MAX_LEVENSHTEIN_COMPARISONS = 100;
+
+    private final List<String> sampleMessages;
+    private final LogFileStructure structure;
+
+    SeparatedValuesLogFileStructureFinder(List<String> explanation, String sample, String charsetName, Boolean hasByteOrderMarker,
+                                          CsvPreference csvPreference, boolean trimFields) throws IOException {
+
+        Tuple<List<List<String>>, List<Integer>> parsed = readRows(sample, csvPreference);
+        List<List<String>> rows = parsed.v1();
+        List<Integer> lineNumbers = parsed.v2();
+
+        Tuple<Boolean, String[]> headerInfo = findHeaderFromSample(explanation, rows);
+        boolean isCsvHeaderInFile = headerInfo.v1();
+        String[] csvHeader = headerInfo.v2();
```

Comment:
nit `delineatedHeader` or `separatedValuesHeader`

Reply:
I changed it to just `header`.  `csv` still appears in a few places because SuperCSV has the same problem of using "CSV" in its name while being a generic delimited values parser.